### PR TITLE
Document prometheus authentication metrics

### PR DESCRIPTION
**Pull request by mrogers950**
Here's some text for the prometheus metrics that we capture relating to auth. I stuck this at the end of architecture/additional_concepts/authentication.adoc though I'm not sure if that's the right place (I figured there may be one place to document all of the system metrics).
@ahardin-rh

### CODE REVIEW

File: architecture/additional_concepts/authentication.adoc [hunk 1]
```diff
@@ -674,3 +674,13 @@ responses from `/oauth/authorize`, and how they should be handled:
 |401    | `WWW-Authenticate` header missing                                                                                                                | No challenge authentication is possible. Fail and show response body (which might contain links or details on alternate methods to obtain an OAuth token)
 |Other  | Other                                                                                                                                            | Fail, optionally surfacing response body to the user
 |===
+
+[[authentication-prometheus-system-metrics]]
+=== Authentication metrics for Prometheus
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authentication.adoc]: Metrics

File: architecture/additional_concepts/authentication.adoc [hunk 2]
```diff
@@ -674,3 +674,13 @@ responses from `/oauth/authorize`, and how they should be handled:
...
+
+OpenShift captures the following Prometheus system metrics duing authentication attempts:
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authentication.adoc]: s/OpenShift/{product-title}

File: architecture/additional_concepts/authentication.adoc [hunk 3]
```diff
@@ -674,3 +674,13 @@ responses from `/oauth/authorize`, and how they should be handled:
...
+* `openshift_auth_basic_password_count` counts the number of `oc login` username and password attempts
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authentication.adoc]: user name

File: architecture/additional_concepts/authentication.adoc [hunk 4]
```diff
@@ -674,3 +674,13 @@ responses from `/oauth/authorize`, and how they should be handled:
...
+* `openshift_auth_basic_password_count_result` counts the number of `oc login` username and password attempts by result (success or error)
+* `openshift_auth_form_password_count` counts the number of web console login attempts
+* `openshift_auth_form_password_count_result` counts the number of web console login attempts by result (success or error)
+* `openshift_auth_password_total` counts the total number of `oc login` and web console login attempts
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authentication.adoc]: period after all bullet points

Comment (ahardin-rh) [inline architecture/additional_concepts/authentication.adoc]: @mrogers950 Just a few comments from me. Otherwise, looks good! Thanks for this!

@bmcelvee I think you're helping to write up some content regarding Prometheus. Is this related to that? If so, do you have any opinions on the placement of this content? Thanks!

**Comment by mrogers950** [comment]
>@mrogers950 Just a few comments from me. Otherwise, looks good! Thanks for this!
>
>@bmcelvee I think you're helping to write up some content regarding Prometheus. Is this related to that? If so, do you have any opinions on the placement of this content? Thanks!

updated.

**Comment by bmcelvee** [comment]
@mrogers950 I am drafting some content for Prometheus. The bulk of the content I'm adding will go into Enabling Cluster Metrics (https://docs.openshift.com/container-platform/3.6/install_config/cluster_metrics.html), but I think this section works here. Thanks!

**Comment by ahardin-rh** [comment]
@mrogers950 Thank you! Changes LGTM.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../architecture/additional_concepts/authentication.adoc#architecture-additional-concepts-authentication[Additional Concepts -> Authentication]
|Added a section on xref:../architecture/additional_concepts/authentication.adoc#authentication-prometheus-system-metrics[Authentication Metrics for Prometheus].
%

File: architecture/additional_concepts/authentication.adoc [hunk 5]
```diff
@@ -674,3 +674,13 @@ responses from `/oauth/authorize`, and how they should be handled:
...
+
+[[authentication-prometheus-system-metrics]]
+=== Authentication Metrics for Prometheus
+
+{product-title} captures the following Prometheus system metrics duing authentication attempts:
+* `openshift_auth_basic_password_count` counts the number of `oc login` user name and password attempts.
+* `openshift_auth_basic_password_count_result` counts the number of `oc login` user name and password attempts by result (success or error).
+* `openshift_auth_form_password_count` counts the number of web console login attempts.
+* `openshift_auth_form_password_count_result` counts the number of web console login attempts by result (success or error).
+* `openshift_auth_password_total` counts the total number of `oc login` and web console login attempts.
```
Comment (sferich888) [inline architecture/additional_concepts/authentication.adoc]: Where are these loged, and/or how are these accessed?

Comment (sferich888) [reply architecture/additional_concepts/authentication.adoc]: I also don't think the names here are intuitive to what they count.
